semgrep/tests/e2e/test_cli_test.py

```diff
@@ -8,6 +8,20 @@ def test_cli_test_basic(run_semgrep_in_tmp, snapshot):
         target_name="cli_test/basic/",
         output_format=OutputFormat.JSON,
     )
+
+    snapshot.assert_match(
+        results,
+        "results.json",
+    )
+
+
+def test_timeout(run_semgrep_in_tmp, snapshot):
+    results, _ = run_semgrep_in_tmp(
+        "rules/cli_test/error/",
+        options=["--test"],
+        target_name="cli_test/error/",
+        output_format=OutputFormat.JSON,
+    )
     snapshot.assert_match(
         results,
         "results.json",
```

Comment:
Maybe reduce the timeout to speed up the tests?
```
    --timeout INTEGER             Maximum time to spend running a rule on a
                                  single file in seconds. If set to 0 will not
                                  have time limit. Defaults to 30 s.
```

Reply:
hm I don't think `--timeout` affects `--test`

Reply:
Anyway I think it stack overflows pretty quickly